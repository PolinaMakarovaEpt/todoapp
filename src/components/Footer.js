import React, { Component } from 'react';

class Footer extends Component {

    handleFilterChanged(e) {
        this.props.onFilterChanged(e.currentTarget.dataset.value);
    }

    render() {
        return (
            <div className="footer">
                <div>
                    <span className="active-items">Активных дел: {this.props.tasks.filter((t) => !t.checked).length}</span>
                </div>

                <div className="footer-buttons">

                    <button className={this.props.filter === "all" ? "selected" : ""}
                        data-value="all"
                        onClick={this.handleFilterChanged.bind(this)}>Все</button>

                    <button className={this.props.filter === "active" ? "selected" : ""}
                        data-value="active"
                        onClick={this.handleFilterChanged.bind(this)}>Активные</button>

                    <button className={this.props.filter === "completed" ? "selected" : ""}
                        data-value="completed"
                        onClick={this.handleFilterChanged.bind(this)}>Завершенные</button>

                </div>

                <div className="delete-completed">
                    <span onClick={this.props.onClearCompleted}>Удалить завершенные</span>
                </div>
            </div>
        )
    }
}


export default Footer;