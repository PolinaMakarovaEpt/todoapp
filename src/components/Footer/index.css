.footer {
    position: relative;
    display: flex;
    width: 100%;
    height: 2rem;
}

.active-items {
    position: relative;
    left: 0.5rem;
    color: rgb(70, 70, 70);
}

.footer-buttons {
    position: relative;
    margin: 0 auto;
}
.footer-buttons button {
    font-size: 17px;
    padding: 0.2rem;
    border: 1px solid white;
    background-color: white;
    outline: none;
    color: rgb(70, 70, 70);
}

.footer-buttons button:hover {
    border: 1px solid rgb(223, 223, 223);
}

.footer-buttons .selected {
    margin-right: 0.1rem;
    border: 1px solid rgb(175, 175, 175);
    outline: none;
    background-color: white;
}

.delete-completed {
    position: relative;
    right: 0.5rem;
    color: rgb(70, 70, 70);
}
.delete-completed:hover {
    color: red;
    text-decoration: underline;
}

@media only screen and  (max-width: 800px) and (min-width: 510px) {
    .footer {
        position: relative;
        font-size: 15px;
        top: 0.5rem;
    }
    .footer button {
        position: relative;
        font-size: 13px;
        bottom: 0.5rem;
    }
    .delete-completed {
        position: relative;
        bottom: 0.5rem;
        width: 20%;
        word-break: keep-all;
        margin-left: 0.2rem;
    }
    .active-items {
        position: relative;
        bottom: 0.5rem;
        margin-right: 0.2rem;
    }
}

@media only screen and  (max-width: 510px) {
    .footer {
        position: relative;
        font-size: 15px;
        top: 0.5rem;
    }
    .footer-buttons {
        width: 30%;
    }
    .footer button {
        position: relative;
        font-size: 13px;
        bottom: 0.5rem;
    }
    .delete-completed {
        position: relative;
        bottom: 0.5rem;
        right: 1rem;
        width: 20%;
        word-break: keep-all;
        margin: 0 0.2rem;
    }
    .active-items {
        position: relative;
        bottom: 0.5rem;
        margin-right: 0.2rem;
    }
}