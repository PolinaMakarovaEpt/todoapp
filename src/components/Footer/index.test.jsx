import React from "react";
import Footer from "./index";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import { JestEnvironment } from "@jest/environment";

configure({ adapter: new Adapter() });

it("matches the snapshot", () => {
    const props = {
        onFilterChanged: jest.fn(),
        tasks: [{
            id: 1,
            title: "",
            checked: false
        }],
        filter: "all",
        onClearCompleted: jest.fn()
    }
    const tree = shallow(<Footer {
        ...props
    } />);
    expect(toJson(tree)).toMatchSnapshot();
});
