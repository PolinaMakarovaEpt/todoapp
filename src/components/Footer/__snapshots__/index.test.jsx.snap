// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="footer"
>
  <div>
    <span
      className="active-items"
    >
      Активных дел: 
      1
    </span>
  </div>
  <div
    className="footer-buttons"
  >
    <button
      className="selected"
      data-value="all"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      className=""
      data-value="active"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      className=""
      data-value="completed"
      onClick={[Function]}
    >
      Завершенные
    </button>
  </div>
  <div
    className="delete-completed"
  >
    <span
      onClick={[MockFunction]}
    >
      Удалить завершенные
    </span>
  </div>
</div>
`;
